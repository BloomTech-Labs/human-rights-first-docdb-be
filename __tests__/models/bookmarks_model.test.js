const db = require('../../data/db-config');
const Bookmarks = require('../../api/bookmarks/bookmarksModel');

beforeAll(async () => {
  await db.migrate.rollback();
  await db.migrate.latest();
});

beforeEach(async () => {
  await db.seed.run();
});

afterAll(async () => {
  await db.destroy();
});

it('sanity test', () => {
  expect(true).toBe(true);
});

describe('getBookmarks', () => {
  let res;
  beforeEach(async () => {
    res = await Bookmarks.getBookmarks();
  });

  it('returns 19 bookmarks', () => {
    expect(res).toHaveLength(19);
  });

  it('returns the correct data shape', () => {
    expect(res).toMatchSnapshot();
  });
});

describe('getByUserId(id)', () => {
  let res;
  const id = 'abc123';
  beforeEach(async () => {
    res = await Bookmarks.getByUserId(id);
  });

  it('returns correct bookmarks of specific user', () => {
    expect(res).toHaveLength(3);
  });

  it('returns the correct data shape', () => {
    expect(res).toMatchSnapshot();
  });
});

describe('addBookmark(bookmark)', () => {
  let res;
  const newBookmark = {
    bookmarkId: -119,
    id: '00ultx74kMUmEW8054x6',
    fileId: '123456789',
  };

  beforeEach(async () => {
    res = await Bookmarks.addBookmark(newBookmark);
  });

  it('returns 20 bookmarks', async () => {
    const getAll = await db('bookmarks');

    expect(getAll).toHaveLength(20);
    expect(getAll[19]).toMatchObject(newBookmark);
  });

  it('returns the correct data shape', () => {
    expect(res).toMatchSnapshot();
  });
});
