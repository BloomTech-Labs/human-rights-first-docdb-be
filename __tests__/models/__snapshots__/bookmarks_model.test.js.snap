// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBookmarks returns the correct data shape 1`] = `
Array [
  Object {
    "bookmarkId": -100,
    "fileId": "1",
    "id": "abc123",
  },
  Object {
    "bookmarkId": -101,
    "fileId": "2",
    "id": "abc123",
  },
  Object {
    "bookmarkId": -102,
    "fileId": "6",
    "id": "abc123",
  },
  Object {
    "bookmarkId": -103,
    "fileId": "1",
    "id": "23jsbb",
  },
  Object {
    "bookmarkId": -104,
    "fileId": "3",
    "id": "23jsbb",
  },
  Object {
    "bookmarkId": -105,
    "fileId": "4",
    "id": "j3iasd",
  },
  Object {
    "bookmarkId": -106,
    "fileId": "9",
    "id": "jsh123",
  },
  Object {
    "bookmarkId": -107,
    "fileId": "8",
    "id": "jsh123",
  },
  Object {
    "bookmarkId": -108,
    "fileId": "7",
    "id": "jsh123",
  },
  Object {
    "bookmarkId": -109,
    "fileId": "2",
    "id": "bnsdj2",
  },
  Object {
    "bookmarkId": -110,
    "fileId": "4",
    "id": "jpe12f",
  },
  Object {
    "bookmarkId": -111,
    "fileId": "6",
    "id": "jpe12f",
  },
  Object {
    "bookmarkId": -112,
    "fileId": "2",
    "id": "jpe12f",
  },
  Object {
    "bookmarkId": -113,
    "fileId": "9",
    "id": "jpe12f",
  },
  Object {
    "bookmarkId": -114,
    "fileId": "8334132410",
    "id": "00ultx74kMUmEW8054x6",
  },
  Object {
    "bookmarkId": -115,
    "fileId": "168448015060",
    "id": "00ultx74kMUmEW8054x6",
  },
  Object {
    "bookmarkId": -116,
    "fileId": "844527667744",
    "id": "00ultx74kMUmEW8054x6",
  },
  Object {
    "bookmarkId": -117,
    "fileId": "8334297344",
    "id": "00ultx74kMUmEW8054x6",
  },
  Object {
    "bookmarkId": -118,
    "fileId": "166144018989",
    "id": "00ultx74kMUmEW8054x6",
  },
]
`;

exports[`getByUserId(id) returns the correct data shape 1`] = `
Array [
  Object {
    "bookmarkId": -100,
    "fileId": "1",
    "id": "abc123",
  },
  Object {
    "bookmarkId": -101,
    "fileId": "2",
    "id": "abc123",
  },
  Object {
    "bookmarkId": -102,
    "fileId": "6",
    "id": "abc123",
  },
]
`;
